src/controllers/user.controllers.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response } from "express";
+import bcrypt from "bcrypt";
+import * as jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+import User from "../models/user";
+import { PayloadAttributes, UserAttributes } from "../types/user.types";
+
+dotenv.config();
+
+const userSignup = async (req: Request, res: Response) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+    const salt = await bcrypt.genSalt(12);
```

Comment:
Create a utility function that encrypts the  password/text and use it here because it might be required anywhere else. Don't forget a decrypt function as well.